crates/valence_inventory/src/lib.rs

```diff
@@ -659,6 +675,20 @@ fn update_player_inventories(
     }
 }
 
+/// Handles the `HeldItem` component being changed on a client, which
+/// indicates that the server has changed the selected hotbar slot.
+fn update_player_selected_slot(mut clients: Query<(&mut Client, &mut HeldItem)>) {
```

Comment:
I would rather use bevy change detection to detect when `HeldItem` is changed, and bypass the change detection when the c2s packet is received.

Reply:
how am I so bad I explaining my thought when it was this easy.